zipline/finance/controls.py

```diff
@@ -137,8 +151,11 @@ def validate(self,
         """
         Fail if the asset is in the restricted_list.
         """
-        if asset in self.restricted_list:
-            self.fail(asset, amount, _algo_datetime)
+        if self.restricted_list.is_restricted(asset, _algo_datetime):
```

Comment:
The underscores here are intended to signify unused parameters, so this should probably no longer be prefixed with an underscore.